EIPS/eip-draft-whisper.md

```diff
@@ -0,0 +1,138 @@
+## Preamble
+
+    EIP: draft
+    Title: Whisper Specification
+    Author: Vlad Gluhovsky <gluk256@gmail.com>
+    Type: Informational
+    Status: Draft
+    Created: 2017-05-05
+
+## Abstract
+
+This draft EIP describes the format of Whisper messages within the ÐΞVp2p Wire Protocol.
+This EIP should substitute the [existing specification](https://github.com/ethereum/wiki/wiki/Whisper-Wire-Protocol).
+More detailed documentation on Whisper could be found [here](https://github.com/ethereum/go-ethereum/wiki/Whisper).
+
+## Motivation
+
+It is necessary to specify the standard for Whipser messages in order to ensure forward compatibility of different Whisper clients.
+
+## Specification
+
+All Whisper messages sent as ÐΞVp2p Wire Protocol packets should be RLP-encoded arrays of data containing two objects: integer packet code followed by another object (whose type depends on the packet code). 
+
+If Whisper node does not support a particular packet code, it should just ignore the packet without generating any error.
+
+### Packet Codes
+
+The Whisper sub-protocol should support the following packet codes:
+
+| EIP   | Name                       | Int Value |
+|-------|----------------------------|-----------|
+|       | Status                     |   0x00    |
+|       | Messages                   |   0x01    |
+|       | P2P                        |   0x02    |
+|       | P2P Request                |   0x03    |
+|       | PoW Requirement            |   0x04    |
+|       | Bloom Filter               |   0x05    |
+
+
+### Packet Format and Usage
+
+**Status** [`0x00`, `whisper_protocol_version`] 
+
+This packet contains two objects: integer (0x00) followed by integer (Whisper protocol version).
+
+Informs a peer of the Whisper version. This message should be send after the initial handshake and prior to any other messages.
+
+**Messages** [`0x01`, `whisper_envelope`]
+
+This packet contains two objects: integer (0x01) followed by a single Whisper Envelope.
+
+This packet is used for sending the standard Whisper envelopes.
+
+**P2P** [`0x02`, `whisper_envelope`]
+
+This packet contains two objects: integer (0x02) followed by a single Whisper Envelope.
+
+This packet is used for sending the peer-to-peer messages, which are not supposed to be forwarded any further. E.g. it might be used by the Whisper Mail Server for delivery of old (expired) messages, which is otherwise not allowed.
+
+**P2P Request** [`0x03`, `whisper_envelope`]
+
+This packet contains two objects: integer (0x03) followed by a single Whisper Envelope.
+
+This packet is used for sending Dapp-level peer-to-peer requests, e.g. Whisper Mail Client requesting old messages from the Whisper Mail Server.
+
+**PoW Requirement** [`0x04`, `PoW`]
+
+This packet contains two objects: integer (0x04) followed by a single floating point value of PoW.
```

Comment:
perhaps an even better metric would be to split the hash into two bit vectors: `leading_zeros` and `nonzero_part` and then define

```
fn work_to_space_time_ratio(pow_hash, size, ttl)
    = (2**len(leading_zeros) + (nonzero_part as uint)) / (size * ttl)
```

to avoid ramp-up being purely exponential